.teamcity/hosted/arm/PushStagingLinux2004_Aarch64.kts

```diff
@@ -0,0 +1,75 @@
+package generated
+
+import common.TeamCityDockerImagesRepo
+import hosted.utils.ImageInfoRepository
+import hosted.utils.steps.buildAndPublishImage
+import jetbrains.buildServer.configs.kotlin.v2019_2.AbsoluteId
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.FailureAction
+import jetbrains.buildServer.configs.kotlin.v2019_2.ReuseBuilds
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.dockerSupport
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.freeDiskSpace
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.swabra
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.dockerCommand
+
+/**
+ * Building and deploying aarch64 (ARM) Linux-based Docker images into staging registry, which is defined ...
+ * ... within upstream's parameters.
+ */
+object push_staging_linux_2004_aarch64 : BuildType({
+    name = "[aarch64] [Linux 2004] Build and deploy staging images"
+    buildNumberPattern = "%dockerImage.teamcity.buildNumber%-%build.counter%"
+    description = "Build and deploy Linux-based Docker images for aarch64 platform."
+    vcs {
+        root(TeamCityDockerImagesRepo.TeamCityDockerImagesRepo)
+    }
+
+    steps {
+        dockerCommand {
+            name = "pull ubuntu:20.04"
+            commandType = other {
+                subCommand = "pull"
+                commandArgs = "ubuntu:20.04"
+            }
+        }
+
+        ImageInfoRepository.getArmImages().forEach { imageInfo ->
+            buildAndPublishImage(imageInfo)
+        }
+    }
+
+    features {
+        freeDiskSpace {
+            requiredSpace = "8gb"
+            failBuild = true
+        }
+
+        dockerSupport {
+            cleanupPushedImages = true
+            loginToRegistry = on {
+                dockerRegistryId = "PROJECT_EXT_774,PROJECT_EXT_315"
```

Comment:
BTW, which two registry we're using here? The staging on the docker hub + internal in Space?

Reply:
```
The staging on the docker hub + internal in Space?
```
@burnasheva yes. Recently, there were multiple cases when team members wanted to publish the built images into Space registry. Having a consistent connection to Space registry would allow to do that.